docs/whats-new/dotnet-7-docs.md

```diff
@@ -0,0 +1,295 @@
+  - Add two new SYSLIB warnings
+  - XmlSecureResolver breaking change
+  - Add SYSLIB0041
+  - Add SYSLIB0040
+  - Add syslib0038 and syslib0039
+- [SYSLIB0012: Assembly.CodeBase and Assembly.EscapedCodeBase are obsolete](../fundamentals/syslib-diagnostics/syslib0012.md) - Add two new SYSLIB warnings
+- [SYSLIB0043: ECDiffieHellmanPublicKey.ToByteArray is obsolete](../fundamentals/syslib-diagnostics/syslib0043.md)
+  - XmlSecureResolver breaking change
+  - Add SYSLIB0041
+- [Overview of .NET source code analysis](../fundamentals/code-analysis/overview.md) - Merge .NET 7 release branch into main
+- [Code quality rules](../fundamentals/code-analysis/quality-rules/index.md)
+  - Add docs for CA2019 and CA2259
+  - Add docs for CA1420 and CA1422
+  - Add doc for CA2260
+  - Add documentation for rule CA1854
+  - Add document for CA1851
+- [Reliability rules](../fundamentals/code-analysis/quality-rules/reliability-warnings.md) - Add docs for CA2019 and CA2259
+- [Usage rules](../fundamentals/code-analysis/quality-rules/usage-warnings.md)
+  - Add docs for CA2019 and CA2259
+  - Add doc for CA2260
+- [CA1416: Validate platform compatibility](../fundamentals/code-analysis/quality-rules/ca1416.md) - Add docs for CA1420 and CA1422
+- [CA1418: Validate platform compatibility](../fundamentals/code-analysis/quality-rules/ca1418.md) - Add docs for CA1420 and CA1422
+- [Portability and interoperability rules](../fundamentals/code-analysis/quality-rules/interoperability-warnings.md) - Add docs for CA1420 and CA1422
+- [Code-style rule options](../fundamentals/code-analysis/code-style-rule-options.md) - Add missing code-style rules
+- [Expression-level preferences](../fundamentals/code-analysis/style-rules/expression-level-preferences.md) - Add missing code-style rules
+- [Use local function instead of lambda (IDE0039)](../fundamentals/code-analysis/style-rules/ide0039.md) - Add missing code-style rules
+- [Code-style rules](../fundamentals/code-analysis/style-rules/index.md)
+  - Add missing code-style rules
+  - Add IDE0160 and IDE0161.
+- [Performance rules](../fundamentals/code-analysis/quality-rules/performance-warnings.md)
+  - Add documentation for rule CA1854
+  - Add document for CA1851
+- [Language rules](../fundamentals/code-analysis/style-rules/language-rules.md)
+  - Add missing code-style rules
+  - Add IDE0160 and IDE0161.
+- [Null-checking preferences](../fundamentals/code-analysis/style-rules/null-checking-preferences.md) - Add missing code-style rules
+- [Pattern matching preferences](../fundamentals/code-analysis/style-rules/pattern-matching-preferences.md) - Add missing code-style rules
+- [.NET runtime contention events](../fundamentals/diagnostics/runtime-contention-events.md) - Update the doc with the new version of ContentionStart
+- [Error codes returned by package validation](../fundamentals/package-validation/diagnostic-ids.md)
+  - add docs for diagnostics CP0012 and CP0013
+  - add docs for diagnostics CP0010 and CP0011
```

Comment:
A lot more diagnostic updates here. Is there an overview page that would be useful?

Reply:
This is an overview page already.